scalyr_agent/copying_manager/copying_manager.py

```diff
@@ -99,12 +99,16 @@ def __init__(self, config, api_key_config):
                 # Important for the understanding. When the shared_object_manager is started, it creates a new process.
                 # We use this process as a process for the worker.
 
+                # set the parent process pid for the SharedObjectManager's watchdog
+                # which should detect if parent was killed.
+                SharedObjectManager.parent_pid = os.getpid()
+
+                # change agent log path for the new worker.
+                # this initializer function will be invoked in the worker's process.
                 worker_agent_log_path = os.path.join(
                     self.__config.agent_log_path, "agent-%s.log" % worker_id
                 )
 
-                # change agent log path for the new worker.
-                # this initializer function will be invoked in the worker's process.
                 def initializer():
                     self._change_worker_process_agent_log_path(worker_agent_log_path)
```

Comment:
Add a new call here `self._start_watchdog(ppid)`  I think that should be invoked after the `_change_worker_process_agent_log_path` call.

I'm suggesting this because it is more consistent with how the main agent process configures the worker processes.  We already are using the `initializer` to allow changes to be made.  We should just use that same approach for this as well.

Reply:
I agree that an initializer is the right place for that. But in our case, it reveals too many internals of the `multiprocesing.SyncManager` library and other stuff making the `copying_manager.py`  even more bloated. 
As I tried to explain here https://github.com/scalyr/scalyr-agent-2/pull/667#issuecomment-743477423, now `multiprocesing.SyncManager` solves its own problems in its own place.